src/main/java/uk/gov/hmcts/darts/authentication/config/AuthenticationAspect.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.darts.authentication.config;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.ws.soap.SoapHeaderElement;
+import uk.gov.hmcts.darts.authentication.component.SoapHeaderConverter;
+import uk.gov.hmcts.darts.cache.token.TokenRegisterable;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.springframework.web.context.request.RequestAttributes.SCOPE_REQUEST;
+
+@Aspect
+@Configuration
+@RequiredArgsConstructor
+@Slf4j
+public class AuthenticationAspect {
+
+    private static final String ACCESS_TOKEN_REQUEST_ATTR = "access_token";
+
+    private final SoapHeaderConverter soapHeaderConverter;
+    private final TokenRegisterable tokenRegisterable;
+
+    @Pointcut("within(uk.gov.hmcts.darts.ws.DartsEndpoint)")
+    public void withinDartsEndpointPointcut() {
+    }
+
+    @Before("withinDartsEndpointPointcut()")
+    public void handleSoapHeaderAuthenticationAdvice(JoinPoint joinPoint) {
+        Optional<Object> soapHeaderOptional = Arrays.stream(joinPoint.getArgs())
+            .filter(soapHeader -> soapHeader instanceof SoapHeaderElement)
+            .findFirst();
+
+        soapHeaderOptional.ifPresent(soapHeaderElement -> soapHeaderConverter.convertSoapHeader((SoapHeaderElement) soapHeaderElement)
```

Comment:
Not sure what an aspect gives us here. We can do everything we need to in isolation from any endpoint

Reply:
Maybe a spring servlet filter or a spring interceptor is more applicable

Reply:
The Aspect here was intended to be a way for me to simply convert the exposed SoapHeaderElement from the DartsEndpoint into the JAXB ServiceContext that we appear to be using in the JwtCache to createToken.
It was to avoid having to put this inside each endpoint operation as we discussed the other day, we hoped an aspect could work for us.

Reply:
Done in commit 394fb93 - Replace @SoapHeader in DartsEndpoint and AuthenticationAspect with SoapRequestInterceptor